schema-util/bigquery/src/main/java/io/apicurio/registry/rules/compatibility/BigqueryCompatibilityChecker.java

```diff
@@ -0,0 +1,64 @@
+package io.apicurio.registry.rules.compatibility;
+
+import io.apicurio.registry.content.ContentHandle;
+import io.apicurio.registry.rules.BigqueryGsonBuilder;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class BigqueryCompatibilityChecker extends BigqueryGsonBuilder implements CompatibilityChecker {
+    @Override
+    public CompatibilityExecutionResult testCompatibility(CompatibilityLevel compatibilityLevel, List<ContentHandle> existingArtifacts, ContentHandle proposedArtifact) {
+        // existingArtifacts is implemented as a LazyContentList, which does not support the spliterator interface.
+        // This means we cannot call stream() on it in java version 11 (does work in version 17)
+        final List<String> existingArtifactsContent = new ArrayList<>();
+        existingArtifacts.forEach(ch -> existingArtifactsContent.add(ch.content()));
+        return testCompatibility(compatibilityLevel, existingArtifactsContent,
+                proposedArtifact.content());
+    }
+
+    @Override
+    public CompatibilityExecutionResult testCompatibility(CompatibilityLevel compatibilityLevel, List<String> existingArtifacts, String proposedArtifact) {
+        if (compatibilityLevel.equals(CompatibilityLevel.NONE)) {
+            return CompatibilityExecutionResult.compatible();
+        }
+        ComparableSchema proposedSchema = toSchema(proposedArtifact);
+        Set<CompatibilityDifference> differences = new HashSet<>();
+        switch (compatibilityLevel) {
+            case FORWARD: {
+                toSchema(existingArtifacts.get(0)).checkCompatibilityWith(proposedSchema, differences);
```

Comment:
I think the latest artifact should be the last in the list. See https://github.com/kvanbrabant/apicurio-registry/blob/main/schema-util/json/src/main/java/io/apicurio/registry/rules/compatibility/JsonSchemaCompatibilityChecker.java#L53

@carlesarnal Can you please confirm the above? Looking at the lazy list, it uses this query https://github.com/kvanbrabant/apicurio-registry/blob/main/app/src/main/java/io/apicurio/registry/storage/impl/sql/CommonSqlStatements.java#L257 which is unordered and can be a bug

Reply:
Sounds like a bug, yes. I'll check.

Reply:
You are probably right. I just never tested with more than one schema version already present.